sdk/typescript/src/types/objects.ts

```diff
@@ -200,19 +200,54 @@ export type Order = 'ascending' | 'descending';
 export function getSuiObjectData(
   resp: SuiObjectResponse,
 ): SuiObjectData | undefined {
-  return resp.status !== 'Exists' ? undefined : (resp.details as SuiObjectData);
+  return resp.data !== null ? (resp.data as SuiObjectData) : undefined;
+}
+
+function parseObjectIdFromString(input: string): string | null {
+  const objectIdRegex = /Object not exists: (\S+)/;
+  const match = input.match(objectIdRegex);
+  return match ? match[1] : null;
+}
+
+function parseSuiObjectRefFromString(input: string): SuiObjectRef | null {
+  const refRegex = /Object deleted: (\S+), version: (\S+), digest: (\S+)/;
+  const match = input.match(refRegex);
+
+  if (match) {
+    return {
+      objectId: match[1],
+      version: match[2],
+      digest: match[3],
+    };
+  } else {
+    return null;
+  }
 }
 
 export function getObjectDeletedResponse(
   resp: SuiObjectResponse,
 ): SuiObjectRef | undefined {
-  return resp.status !== 'Deleted' ? undefined : (resp.details as SuiObjectRef);
+  if (resp.error !== null) {
+    const objRef = parseSuiObjectRefFromString(resp.error!);
+    if (objRef !== null) {
+      return objRef as SuiObjectRef;
+    }
+  }
+
+  return undefined;
 }
 
 export function getObjectNotExistsResponse(
   resp: SuiObjectResponse,
 ): ObjectId | undefined {
-  return resp.status !== 'NotExists' ? undefined : (resp.details as ObjectId);
+  if (resp.error !== null) {
+    const obj_id = parseObjectIdFromString(resp.error!);
```

Comment:
Absolutely hate this 😅 Lets not parse error messages, and instead just return the data we need.

Reply:
changed this to parse the error response structure